src/entities/components/Rotator.ts

```diff
@@ -0,0 +1,43 @@
+import { vec2 } from 'gl-matrix'
+
+import { TransformData } from '~/interfaces'
+import { getAngle } from '~/util/math'
+
+const normalizeAngle = (theta: number): number => {
+  if (theta > Math.PI) {
+    return theta - 2 * Math.PI
+  } else if (theta < -Math.PI) {
+    return theta + 2 * Math.PI
+  }
+  return theta
+}
+
+export const getAngularDistance = (params: {
+  from: TransformData
+  to: vec2 | number
+}): number => {
+  const { from, to } = params
+  const targetOrientation =
+    typeof to === 'number' ? to : getAngle(from.position, to)
+  return normalizeAngle(targetOrientation - from.orientation)
+}
+
+export class Rotator {
+  speed: number
+
+  constructor(params: { speed: number }) {
+    this.speed = params.speed
+  }
+
+  rotate(params: { from: TransformData; to: vec2 | number; dt: number }) {
```

Comment:
I wonder how much this needs to be a class, as opposed to a function that takes an additional param `speed`, or even just even `angularDisplacement` (which is just `dt` * `speed`)? It doesn't seem like encapsulating the speed benefits either of the use cases it's being used for.

Reply:
Yeah, I had the same thought after finishing as well. I might do another refactor pass.